src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -93,7 +111,12 @@ public ReadsDataSource( final List<File> samFiles ) {
         backingFiles = new LinkedHashMap<>(samFiles.size() *2);
         indicesAvailable = true;
 
-        for ( File samFile : samFiles ) {
+        final SamReaderFactory SAMReaderFactory =
+                samReaderFactory == null ?
+                    SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT) :
+                    samReaderFactory;
```

Comment:
Could you improve the variable naming here? You have both `samReaderFactory` and `SAMReaderFactory` in this method, which is confusing. How about `customSAMReaderFactory` for the constructor argument, and `samReaderFactory` for the factory that actually gets used?